lib/MockXMLHttpRequest.js

```diff
@@ -158,40 +159,45 @@ MockXMLHttpRequest.prototype.send = function(data) {
 
     var response = MockXMLHttpRequest.handle(new MockRequest(self));
 
-    if (response && response instanceof MockResponse) {
+    Promise.resolve(response).then(function(response) {
```

Comment:
With the current implementation of `MockXMLHttpRequest.handle`, handlers must return a `falsey` value in order to signify they don't want to handle the request.

That means if any of the handlers returns a `Promise` (which is `truthy`), and the `Promise` does not `resolve` to a response, the `MockXMLHttpRequest.handle` mehtod won't continue to loop through the other handlers and result in error because no response was returned, even if there was another handler that would have returned a result.

e.g.
```js
function handler(req, res) {
  return new Promise(function(resolve, reject) {
    //resolving without a response if we decided to not handle the request
    setTimeout(resolve, 100);
  });
}
```

This feels like an edge case (creating a handler that asynchronously decides not to return a response) but a bug none-the-less. I wonder if we should handle this in the `MockXMLHttpRequest.handle` method?

e.g. 
```js
//untested code:
MockXMLHttpRequest.handle = function(request) {
  var promise = Promise.resolve();
  for (var i=0; i<MockXMLHttpRequest.handlers.length; ++i) {
      promise = promise.then(function(response) {

        //if one of the previous handlers has already returned a response, use that
        if (response) {
          return response;
        }

        return Promise.resolve(MockXMLHttpRequest.handlers[i](request, new MockResponse()));
      });

  }
  return promise;
}
```